batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/utils/requestmapper/RequestMapper.java

```diff
@@ -0,0 +1,213 @@
+package com.lftechnology.batch7crud.utils.requestmapper;
+
+import com.lftechnology.batch7crud.constant.PageConstant;
+import com.lftechnology.batch7crud.controller.AuthenticationController;
+import com.lftechnology.batch7crud.controller.StudentController;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * @Author binodnme
+ * Created on 1/29/16
+ */
+public class RequestMapper {
+  private static final Logger LOGGER = Logger.getLogger(RequestMapper.class.getName());
+
+
+  public void map(HttpServletRequest request, HttpServletResponse response) throws ServletException {
+    String servletPath = request.getServletPath();
+    String pathInfo = request.getPathInfo();
+    String requestType = request.getMethod();
+
+    Class studentController = StudentController.class;
+    Class authenticationController = AuthenticationController.class;
+    List<Class> controllers = new ArrayList<>();
+    controllers.add(studentController);
+    controllers.add(authenticationController);
+
+
+
+    Class controller = filterController(controllers, servletPath);
+
+    if(controller == null){
+      throw new ServletException(PageConstant.PAGE_NOT_FOUND);
+    }
+
+    Method[] methods = controller.getDeclaredMethods();
+    List<Method> methodList = Arrays.asList(methods);
+
+    Method method = filterMethod(methodList, pathInfo, requestType);
+
+
+    if(method != null){
+      Object obj;
+      try {
+        obj = controller.newInstance();
+        method.setAccessible(true);
+        method.invoke(obj, request, response);
+      } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      }
+    }else {
+      throw new ServletException(PageConstant.PAGE_NOT_FOUND);
+    }
+  }
+
+
+  private Class filterController(List<Class> controllers, String servletPath){
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "filterController" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)